test/test_blocks_on_demand.py

```diff
@@ -0,0 +1,106 @@
+"""
+Test blocks on demand mode.
+"""
+
+from test.rpc.rpc_utils import gateway_call
+
+import requests
+
+from .account import invoke
+from .settings import APP_URL
+from .shared import (
+    ABI_PATH,
+    CONTRACT_PATH,
+    PREDEPLOY_ACCOUNT_CLI_ARGS,
+    PREDEPLOYED_ACCOUNT_ADDRESS,
+    PREDEPLOYED_ACCOUNT_PRIVATE_KEY,
+)
+from .util import ReturnCodeAssertionError, call, deploy, devnet_in_background
+
+
+@devnet_in_background(
+    *[
+        *PREDEPLOY_ACCOUNT_CLI_ARGS,
+        "--blocks-on-demand",
+    ]
+)
+def test_blocks_on_demand_invoke():
+    """Test deploy in blocks on demand mode"""
```

Comment:
Could you write it as `blocks-on-demand` (with hyphens) so it's easier to understand.